### PR TITLE
Generate puppet (compute and sql) with The Magician.

### PR DESCRIPTION
**Pull request by ndmckinley**
Begin to generate Puppet (compute and sql) in The Magician.

-----------------------------------------------------------------
# [all]
No Changes
## [puppet]
Initial catch-up PR.

### CODE REVIEW

**Comment by ndmckinley** [comment]
This does seem to work.  But testing puppet is _nuts_.

https://seven.ci.cloud-graphite.com/teams/main/pipelines/magic-modules-testing/jobs/puppet-test/builds/4

@rambleraptor and I took a look at this build (you'll need credentials on the CI testing server - check the credhub instance).  We learned that these tests are wild!  They pass at puppet 5.3.3, which is what's in `Gemfile.lock` right now ... unless combined with openssl 1.1.0 (which happens to be the version in the `ruby:2.5-stretch` container).  That issue was fixed by https://github.com/puppetlabs/puppet/commit/8577b19f4b12b6c43baed68a5d4aed9f87803ed9, which is brought in at 5.3.4.  If we update to the most recent puppet, 5.5.0, the tests fail for some completely different reason to do with Mocha.

I just did some more checking, and I found out that if we update the puppet gem just to 5.3.4 but no further, the tests *mostly* pass.  I'm pushing that now and we'll see what happens.  :)

**Comment by ndmckinley** [comment]
And you can see that succeeding here:
https://seven.ci.cloud-graphite.com/teams/main/pipelines/magic-modules-testing/jobs/puppet-test/builds/6

**Comment by ndmckinley** [comment]
@rambleraptor @nelsonjr 
The puppet compute tests make it through several thousand test cases, all the way to

```
Puppet::Type::Gcompute_instance_template::ProviderGoogle
  #instances
  ensure == present
    resource exists
      no changes == no action
        title == name
          title == name (pass)
            Gcompute_instance_template[title0]
````

before freezing (apparently forever).  Both on my machine and in Concourse.  Do you ever run the `puppet/compute` tests yourselves?  Is this something you know about?

**Comment by ndmckinley** [comment]
Note to reviewers: this is running on the testing server here:
https://seven.ci.cloud-graphite.com/teams/main/pipelines/magic-modules-testing

So the PRs you see there are correct.

File: .ci/magic-modules/create-pr.sh [hunk 1]
```diff
@@ -33,15 +36,59 @@ $(git log -1 --pretty=%B)
...
+    DEPENDENCIES="${DEPENDENCIES}depends: $PUP_COMPUTE_PR ${NEWLINE}"
+  else
+    echo "Puppet Compute - did not generate a PR."
+  fi
+  popd
+
+  if [ -z "$DEPENDENCIES" ]; then
+    cat << EOF > ./pr_comment 
+I am a robot that works on MagicModules PRs!
+I checked the downstream repositories, and none of them seem to have any changes.
```
Comment (rosbo) [inline .ci/magic-modules/create-pr.sh]: Neat feature.

The message however implies that we checked all the downstream repositories. Consider changing the wording and including a link to `.ci/README.md` and update `.ci/README.md` to include the list of downstream repos the magician is checking.

As you add support for more downstream repos, it allows us to easily track what was supported when the PR was created.

Comment (ndmckinley) [reply .ci/magic-modules/create-pr.sh]: Done!

**Pull request merged**